datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -536,7 +553,7 @@ protected int constructExpectedNumberOfTasks(DatastreamGroup dg, List<String> in
     return expectedNumberOfTasks;
   }
 
-  private void updateOrRegisterElasticTaskAssignmentMetrics(DatastreamGroupPartitionsMetadata datastreamPartitions,
+  protected void updateOrRegisterElasticTaskAssignmentMetrics(DatastreamGroupPartitionsMetadata datastreamPartitions,
```

Comment:
This change implies that we will be exposing the following configuration options for rollback and forward deployment.
a) Regular StickyPartitionAssignmentStrategy.
b) StickyPartitionAssignmentStrategy with elastic task count based on partition count.
c) LoadBalancedAssigmentStrategy with elastic task count based on partition count and elastic task count based on throughput info.
Basically if elastic Task count config is enabled with new strategy, it will do based on both partition count and throughput. We don't have knob to disable one calculation. If we want to rollback to just task count based on partition count, we will have to change the strategy itself.

Reply:
I'll document the rollout and rollback instructions separately.